sbi/inference/snpe/snpe_base.py

```diff
@@ -435,6 +423,25 @@ def _train(
         self._summary["epochs"].append(epoch)
         self._summary["best_validation_log_probs"].append(self._best_val_log_prob)
 
+    def _loss(self, round_idx: int, theta: Tensor, x: Tensor, masks: Tensor) -> Tensor:
+        """Return the right calibrated loss, depending on the round.
+
+        The loss is the negative log prob. Irrespective of the round or SNPE method
+        (A, B, or C) it weighted with the calibration kernel.
```

Comment:
*it* is weighted with *a* calibration kernel (the, it is not clear, calibration could happen on the basis of other stuff, not just failed simulations).